File: test_tabak.py
#!/usr/bin/env python3
"""Test script to inspect the Excel file structure"""
import sys

try:
    import openpyxl
    wb = openpyxl.load_workbook(r'C:\Users\sg.msa\Downloads\Şube Bazlı Toplam Tabak Sayısı - 1.02.2026 06_00_00 - 1.03.2026 06_00_00.xlsx', read_only=True)
    print('Sheet names:', wb.sheetnames)
    for sname in wb.sheetnames:
        ws = wb[sname]
        print(f'\n--- Sheet: {sname} ---')
        count = 0
        for i, row in enumerate(ws.rows):
            if i <= 4:  # First 5 rows
                vals = [cell.value for cell in row]
                types = [type(cell.value).__name__ for cell in row]
                print(f'  Row {i+1} types={types}: {vals}')
            count += 1
        print(f'  Total rows iterated: {count}')
    wb.close()
except Exception as e:
    print(f'openpyxl error: {e}')
    import traceback; traceback.print_exc()

File: test_tabak.js
const XLSX = require('xlsx');

const filePath = 'C:\\Users\\sg.msa\\Downloads\\Şube Bazlı Toplam Tabak Sayısı - 1.02.2026 06_00_00 - 1.03.2026 06_00_00.xlsx';

try {
    const workbook = XLSX.readFile(filePath);
    console.log('Sheet names:', workbook.SheetNames);

    for (const name of workbook.SheetNames) {
        const ws = workbook.Sheets[name];
        console.log(`\n--- Sheet: "${name}" ---`);

        // Try default parse
        const rows = XLSX.utils.sheet_to_json(ws, { defval: "" });
        console.log(`  sheet_to_json rows: ${rows.length}`);
        if (rows.length > 0) {
            console.log('  First row keys:', Object.keys(rows[0]));
            console.log('  First row:', JSON.stringify(rows[0]));
        }

        // Try raw parse with header:1 to get array of arrays
        const rawRows = XLSX.utils.sheet_to_json(ws, { header: 1 });
        console.log(`  raw array rows: ${rawRows.length}`);
        if (rawRows.length > 0) {
            console.log('  Raw row 0:', JSON.stringify(rawRows[0]));
        }
        if (rawRows.length > 1) {
            console.log('  Raw row 1:', JSON.stringify(rawRows[1]));
        }
        if (rawRows.length > 2) {
            console.log('  Raw row 2:', JSON.stringify(rawRows[2]));
        }
    }
} catch (e) {
    console.error('Error:', e.message);
}
